Packs/AggregatedScripts/Scripts/QuarantineFile/QuarantineFile.py

```diff
@@ -0,0 +1,1145 @@
+        """
+        Verifies that a file hash is provided and that its type is supported by the target brands.
+
+        Args:
+            brands_to_run (list): The list of active brands that will be used.
+
+        Raises:
+            DemistoException: If the hash argument is missing, the hash type is unsupported,
+                              or no enabled integration supports the given hash type.
+        """
+        if not self.args.get(self.FILE_HASH_ARG):
+            raise DemistoException(f"Missing required argument. Please provide '{self.FILE_HASH_ARG}'.")
+
+        hash_type = get_hash_type(self.args.get(self.FILE_HASH_ARG)).lower()
+        supported_brands_for_hash = self.HASH_TYPE_TO_BRANDS.get(hash_type)
+
+        if not supported_brands_for_hash:
+            raise DemistoException(
+                f"Unsupported hash type: {hash_type}. Supported types are: {', '.join(self.HASH_TYPE_TO_BRANDS.keys())}"
+            )
+
+        if not any(brand in brands_to_run for brand in supported_brands_for_hash):
+            raise DemistoException(
+                "Could not find enabled integrations for the requested hash type.\n"
+                f"For hash_type {hash_type.upper()} please use one of the following brands: "
+                f"{', '.join(supported_brands_for_hash)}"
+            )
+
+    def _sanitize_and_validate_args(self):
+        """
+        Performs all upfront argument validation and sanitization.
+
+        This method orchestrates the various verification checks to ensure the script
+        is running with valid and clean inputs before any actions are taken.
+
+        Raises:
+            DemistoException: If any validation check fails.
+        """
+        demisto.debug("[Orchestrator] Sanitizing and validating script arguments.")
+
+        unique_ids = self._verify_and_dedup_endpoint_ids()
+        self.args[self.ENDPOINT_IDS_ARG] = unique_ids
+
+        brands_to_run = self._verify_and_get_valid_brands()
+        self.args["quarantine_brands"] = brands_to_run
+
+        self._verify_file_hash(brands_to_run)
+
+        demisto.debug("[Orchestrator] Finished sanitizing and validating script arguments.")
+
+    def _is_first_run(self) -> bool:
+        """
+        Determines if this is the first execution of the script for this task.
+
+        Returns:
+            bool: True if there are no pending jobs in the context, False otherwise.
+        """
+        return not self.pending_jobs
+
+    def _job_is_still_polling(self, metadata: dict) -> bool:
+        """
+        Checks the metadata from a command response to see if polling should continue.
+
+        Args:
+            metadata (dict): The 'Metadata' dictionary from a command's raw response.
+
+        Returns:
+            bool: True if the 'polling' flag in the metadata is set to True, False otherwise.
+        """
+        return metadata.get("polling") is True
+
+    def run(self) -> PollResult:
```

Comment:
very happy with how this command turned out, looks great

Reply:
thanks for the direction!!